### PR TITLE
ref(similarity): Add project option to enable ingestion after backfill

### PR DESCRIPTION
**Pull request by jangjodi**
Add enable ingestion flag to backfill endpoint
Add enable ingestion flag in backfill
If the flag is enabled, add `sentry:similarity_backfill_completed` to project options (to be checked in ingestion)

### CODE REVIEW

File: src/sentry/tasks/embeddings_grouping/utils.py [hunk 1]
```diff
@@ -174,6 +176,13 @@ def get_current_batch_groups_from_postgres(project, last_processed_group_id, bat
             "backfill_seer_grouping_records.no_more_groups",
             extra={"project_id": project.id},
         )
+        if enable_ingestion:
+            logger.info(
+                "backfill_seer_grouping_records.enable_ingestion",
+                extra={"project_id": project.id},
+            )
+            project.update_option("sentry:similarity_backfill_completed", int(time.time()))
+
```
Comment (lobsterkatie) [inline src/sentry/tasks/embeddings_grouping/utils.py:185]: Shouldn't we wait to set this option until we know the backfill hasn't errored out?

File: tests/sentry/tasks/test_backfill_seer_grouping_records.py [hunk 2]
```diff
@@ -1534,3 +1539,55 @@ def test_backfill_seer_grouping_records_killswitch_enabled(self, mock_logger):
...
         )
+
+    @with_feature("projects:similarity-embeddings-backfill")
+    @patch("sentry.tasks.embeddings_grouping.utils.logger")
+    @patch("sentry.tasks.embeddings_grouping.utils.post_bulk_grouping_records")
+    def test_backfill_seer_grouping_records_enable_ingestion(
+        self, mock_post_bulk_grouping_records, mock_logger
+    ):
+        """
+        Test that the metadata is set for all groups showing that the record has been created.
```
Comment (lobsterkatie) [inline tests/sentry/tasks/test_backfill_seer_grouping_records.py:1550]: Isn't the point of this test (and the next one) to show what happens with the project option, not with the metadata?

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py [hunk 3]
```diff
@@ -33,15 +33,21 @@ def post(self, request: Request, project) -> Response:
 
         last_processed_id = None
         only_delete = False
+        enable_ingestion = True
```
Comment (trillville) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:36]: is there a good reason to default this to `True`? May be less risky to default false

**Review by trillville — APPROVED** [review]

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py (hunk 3 above)
Comment (jangjodi) [reply src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:36]: I figured we'd typically want to have it be true, but makes sense either way

File: src/sentry/tasks/embeddings_grouping/utils.py (hunk 1 above)
Comment (jangjodi) [reply src/sentry/tasks/embeddings_grouping/utils.py:185]: once we reach this point, we've completed backfilling for that project. we could error out later, but that project has been completed

File: tests/sentry/tasks/test_backfill_seer_grouping_records.py (hunk 2 above)
Comment (jangjodi) [reply tests/sentry/tasks/test_backfill_seer_grouping_records.py:1550]: haha thanks, copy and paste error. I'll fix it

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py (hunk 3 above)
Comment (trillville) [reply src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:36]: yeah unless it greatly simplifies things id prefer we default to false so we can make it more explicit whether or not its happening when we do the backfills

Comment (trillville) [reply src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:36]: (worst case for example we mean to do a test backfill of a random language and brick their ingestion bc we forget to pass the flag)

File: src/sentry/tasks/embeddings_grouping/utils.py (hunk 1 above)
Comment (lobsterkatie) [reply src/sentry/tasks/embeddings_grouping/utils.py:185]: Ah, I see. I missed the `if total_groups_to_backfill_length == 0` which was hidden in the fold.

**Review by lobsterkatie — APPROVED** [review]

**Pull request merged**